### PR TITLE
Basic Status Handler

### PR DESCRIPTION
**Pull request by evangrayk**
Add a Status Request Handler. Currently this displays what handlers have been registered. Still TODO in another PR is to show a record of previous requested URLs and statuses. That change will require a thread-safe multi-map counter, so I figured that can be a separate PR.

Note: the status handler must query the webserver for information. It seemed like a bad idea to use a singleton, in case future use requires multiple server instances. So the status handler only looks for the most recent Webserver Instance in a global. Even though we would possibly support multiple instances, we don't test for it.

### CODE REVIEW

**Comment by evangrayk** [comment]
It looks like this currently:
screen shot 2017-02-23 at 8 11 35 pm

File: src/echo_handler.h [hunk 1]
```diff
@@ -10,6 +10,10 @@ class EchoHandler : public RequestHandler {
                 Response* response);
 
         virtual ~EchoHandler() = default;
+
+        virtual std::string type() const {
```
Comment (Filisha) [inline src/echo_handler.h:14]: Why was this required?

**Review by Filisha — APPROVED** [review]

File: src/echo_handler.h (hunk 1 above)
Comment (evangrayk) [reply src/echo_handler.h:14]: I needed to be able to ask a handler what it's type was, or else the status handler would have no way of showing what type of handler is registered to a certain uri prefix. This seemed like a sane way to extend the API, as it is easy to define for new handler instances and yet gives straight-forward access to aid in logging, etc.
